functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+function Restore-DbaBackup
```

Comment:
@PowerDBAKlaas  - Knocked up a basic flow diagram if that's any help:
[Restore-flow1.pdf](https://github.com/sqlcollaborative/dbatools/files/752417/Restore-flow1.pdf)